modular_skyrat/modules/nuclear_reactor/code/reactor.dm

```diff
@@ -0,0 +1,410 @@
+/*
+	Experimental boiling water reactor
+	This reactor utilises plumbing and atmoshperics to run an internal pressure vessel which generates heat and steam which is transferred into a turbine.
+
+	Reactor layout
+	* = fuel rod
+	- = control rod
+
+	* - *
+	- * -
+	* - *
+
+	The reactor uses incoming gas to cool itself down, and then transfers this very hot gas out to be cooled again. It requires a decent cooling set up to work.
+*/
+
+
+/obj/machinery/reactor
+	name = "Micron Control Systems GA37W Experimental Nuclear Reactor"
+	desc = "A massive experimental boiling water reactor made by Micron Control Systems."
+	icon = 'modular_skyrat/modules/nuclear_reactor/icons/reactor96x96.dmi'
+	icon_state = "idle"
+	bound_x = 96
+	bound_y = 96
+	layer = OBJ_LAYER
+
+	/// Our looping sound for when we're on.
+	var/datum/looping_sound/reactor/soundloop
+	/// We only process once every second.
+	var/next_slowprocess = 0
+	/// Is the reactor cover open?
+	var/cover = FALSE
+	/// The last calculated power output of the reactor.
+	var/calculated_power = 0
+	/// The last calculated control rod efficency
+	var/calculated_control_rod_efficiency = 0
+	/// The last calculated cooling potential in kelvin, it is the incoming gas temperature.
+	var/calculated_cooling_potential = 0
+	/// The current heat of the reactor vessel in kelvin. We start at room temperature.
+	var/core_temperature = T20C
+	/// The core pressure of the reactor vessel in kilopascals.
+	var/core_pressure = 0
+	/// The integrity of the internal reactor core, once this reaches 0, the reactor will explode
+	var/vessel_integrity = 100
+	/// References to currently installed fuel rods.
+	var/list/fuel_rods = list()
+	/// References to currently installed control rods.
+	var/list/control_rods = list()
+	/// What insertion percent do we want the control rods to be at?
+	var/desired_control_rod_insertion = 0
+	/// The currently installed moderator.
+	var/obj/item/reactor_moderator/reactor_moderator
+	/// A reference to our connected heat exchanger.
+	var/obj/machinery/atmospherics/components/binary/reactor_heat_exchanger/reactor_heat_exchanger
+
+/obj/machinery/reactor/Initialize(mapload)
+	. = ..()
+	update_appearance()
+	soundloop = new(src)
+
+/obj/machinery/reactor/update_overlays()
+	. = ..()
+	if(cover)
+		. += "cover"
+
+/obj/machinery/reactor/process()
+	if(next_slowprocess < world.time)
+		reactor_fire()
+		next_slowprocess = world.time + 1 SECONDS
```

Comment:
Use the COOLDOWN system and make sure to early return.

Reply:
A cooldown system for this is not really what we're looking for, since it's more designed for cooldowns and not running processes